enzy_htp/_interface/bcl_interface.py

```diff
@@ -0,0 +1,122 @@
+    in BCL.
+
+
+    Attributes:
+        config_	: The BCLConfig() class which provides settings for both running BCL and maintaining a compatible environment.
+        env_manager_ : The EnvironmentManager() class which ensure all required environment elements exist.
+        compatible_env_ : A bool() indicating if the current environment is compatible with the object itself.
+    """
+
+    def __init__(self, parent, config: BCLConfig = None) -> None:
+        """Simplistic constructor that optionally takes a BCLConfig object as its only argument.
+        Calls parent class.
+        """
+        super().__init__(parent, config, default_bcl_config)
+
+    def _run_bcl_cmd(self, protocol: str, flags: List[str]) -> None:
+        """Underlying method to use the BCL. BCL has a format/calling convention of:
+            $ bcl.exe <protocol> <flags>
+        
+        This function is NOT meant to be used directly but should be used elsewhere in the BCLInterface class.
+
+        Args:
+            protocol: The name of the BCL protocol to use as a str(). 
+            flags: A list() of str() flags to pass to bcl.exe.
+
+        Returns:
+            Nothing.
+
+        """
+        flags = [protocol] + flags
+        self.env_manager_.run_command(self.config_.BCL_EXE, flags)
+
+    def generate_conformers(self, molfile: str, n_conformers: int = 100, outfile: str = None) -> str:
+        """Method that creates conformers using the 'molecule:ConformerGenerator' method from the BCL.
+        Note that the supplied molfile must exist but outfile does not. All conformers are saved to one
+        file. When the outfile is not supplied, the output file is saved to the same directory as the 
+        input file with the suffix '_conformers' added to the name.
+
+        Args:
+            molfile: str() with the name of the input file containing the small molecule of interest. 
+            n_conformers: The int() number of conformers to make.
+            outfile: str() with the path to save the conformers to. Optional argument.
+
+
+        Returns:
+           A str() naming the file with the generated conformers. 
+        """
+
+        #TODO(CJ): add the ability to add more flags and that molfile is in the correct format
+
+        fs.check_file_exists(molfile)
+
+        if not outfile:
+            fpath = Path(molfile)
+            outfile = fpath.parent / f"{fpath.stem}_conformers.sdf"
+
+        flags: List[str] = [
+            "-conformation_comparer",
+            "SymmetryRMSD",
+            "0.25",
+            "-ensemble_filenames",
+            molfile,
+            "-top_models",
+            f"{int(n_conformers)}",
+            "-conformers_single_file",
+            f"{outfile}",
+        ]
+
+        self._run_bcl_cmd("molecule:ConformerGenerator", flags)
+
+        return outfile
+
+    def formal_charge(self, molfile: str) -> int:
```

Comment:
Maybe get_formal_charge is a better name?

Reply:
I disagree. The prefix `get_` just adds length to the function without adding a ton of meaning to the function.

Reply:
My argument is, "get" makes it a verb phrase that the original noun phrase is not so intuitive as a function name (It sounds like formal_charge is a property of BCL unless you know what BCL is).
Also, it helps categorize methods in a class that in auto-complete, typing "get" will list you all the "get" type functions.

Reply:
Ill meet you halfway. `get_` is an equally bad prefix. I'll change it to `calculate_formal_charge()`

Reply:
Changed and committed.